src/motion/features/viewport/types.ts

```diff
@@ -0,0 +1,19 @@
+import { RefObject } from "react"
+import { TargetAndTransition } from "../../../types"
+import { VariantLabels } from "../../types"
+
+export type ViewportEventHandler = () => void
+
+export interface ViewportOptions {
+    root?: RefObject<Element>
+    once?: boolean
+    margin?: string
+    amount?: "some" | "all"
```

Comment:
In `useAppearEffect` we https://github.com/framer/FramerStudio/blob/d6077be8587602ac1a2e8eb901da57a91aaffc30/src/library/src/modules/useAppearEffect.ts#L82-L86, but only pass it 0, 0.5, and 1 as value.

Can we add a `half` or something similar to this?

Reply:
Sure, I wanted to avoid the fractional values for the reasons you were telling me about the other day, but perhaps we could also accept a number as an undocumented type